Rename ChampionButton to CoursesButton

AndrePosm opened the pull request:
Renamed ChampionButton to CoursesButton and updated all related imports to reflect the change.

AndrePosm committed:
Updated button texts for improved clarity

AndrePosm committed:
Renamed ChampionButton.js to CoursesButton.js to match the new button name

VasylPosmit commented:
approved, let's merge it

VasylPosmit commented on src/components/CoursesButton.js:
```diff
@@ -0,0 +1,40 @@
+function CoursesButton() {
+  return (
+    <div style={{ position: 'relative', display: 'inline-block' }}>
+      <Button
```
style = style
onMouseEnter = onMouseEnterHandler
onMouseLeave = onMouseLeaveHandler

VasylPosmit commented on src/components/Header.js:
```diff
@@ -21,7 +21,7 @@ function Header() {
           onMouseLeave={(e) => e.target.style.transform = 'scale(1)'}   // Возврат к исходному размеру
         >
-          GET A HUG
+          BECOME A CHAMPION
```
what does it mean?

better 'get my programm' or 'apply for program'

AndrePosm replied:
I’ve refactored the button component to improve readability and maintainability by moving the hover effects to separate handler functions. This avoids repetition and ensures easier updates in the future.

Let me know if any additional changes are needed.

VasylPosmit replied:
name 'BECOME A CHAMPION' is ambiguous...not clear what it would do

actions (button) must be clear for user... life is hard on it's own

VasylPosmit replied:
so you moved it to Header component

got you 👍

AndrePosm committed:
Refactored button code to simplify hover handling and improve readability

AndrePosm committed:
Refactored CoursesButton to improve code readability and modularity

AndrePosm committed:
Optimized text styles and readability on Home page

AndrePosm commented:
- Renamed buttons for improved clarity.
	- Aligned text on the homepage to the left for better readability.
	- Added indentation for the first lines of paragraphs to enhance text formatting.
	- Cleaned up and optimized the button hover code for improved maintainability and readability.

Let me know if any further changes are needed!

AndrePosm committed:
Refactored button hover effect for better readability and self-explanatory code

AndrePosm commented:
Refactoring Button Hover Effect for Improved Readability

In this update, I have refactored the button hover effect to enhance code readability and maintainability. Instead of generic names like onMouseEnterHandler, I’ve applied more self-explanatory names such as applyHoverEffect and revertHoverEffect. Additionally, I’ve updated the button styles to use a more descriptive constant standardPrimaryButtonStyles, which improves clarity and reduces repetitive code.

These changes align with the best practices of writing self-explanatory code, making it easier for future developers to understand the code’s functionality at a glance.
